nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/test/java/org/apache/nifi/snmp/operations/SNMPRequestTest.java

```diff
@@ -64,53 +82,210 @@
+    private final String setInvalidOidStatusMessage;
+
+    public SNMPRequestTest(final int version, final SNMPConfigurationFactory snmpConfigurationFactory, final TestAgent agent,
+                           final String cannotSetReadOnlyOidStatusMessage, final String cannotModifyOidStatusMessage,
+                           final String getInvalidOidStatusMessage, final String setInvalidOidStatusMessage) {
+        this.version = version;
+        this.snmpConfigurationFactory = snmpConfigurationFactory;
+        this.agent = agent;
+        this.cannotSetReadOnlyOidStatusMessage = cannotSetReadOnlyOidStatusMessage;
+        this.cannotModifyOidStatusMessage = cannotModifyOidStatusMessage;
+        this.getInvalidOidStatusMessage = getInvalidOidStatusMessage;
+        this.setInvalidOidStatusMessage = setInvalidOidStatusMessage;
     }
 
-    protected SNMPTreeResponse getTreeEvents(final int port, final int version) throws IOException {
-        final Snmp snmp = SNMPTestUtils.createSnmpClient();
-        final CommunityTarget target = SNMPTestUtils.createCommTarget("public", LOCALHOST + "/" + port, version);
-        final StandardSNMPRequestHandler standardSnmpRequestHandler = new StandardSNMPRequestHandler(snmp, target);
-        return standardSnmpRequestHandler.walk("1.3.6.1.4.1.32437");
+    @Test
+    public void testSuccessfulSnmpGet() throws IOException {
+        final SNMPConfiguration snmpConfiguration = snmpConfigurationFactory.createSnmpGetSetConfiguration(agent.getPort());
+        try (final SNMPResourceHandler snmpResourceHandler = SNMPFactoryProvider.getFactory(version)
+                .createSNMPResourceHandler(snmpConfiguration)) {
+            final GetSNMPHandler getSNMPHandler = new GetSNMPHandler(snmpResourceHandler);
+            final SNMPSingleResponse response = getSNMPHandler.get(READ_ONLY_OID_1, Collections.emptyMap());
+
+            assertEquals(READ_ONLY_OID_VALUE_1, response.getVariableBindings().get(0).getVariable());
+            assertEquals(SUCCESS, response.getErrorStatusText());
+        }
+    }
+
+    @Test
+    public void testSuccessfulSnmpGetWithFlowFileInput() throws IOException {
+        final SNMPConfiguration snmpConfiguration = snmpConfigurationFactory.createSnmpGetSetConfiguration(agent.getPort());
+        try (final SNMPResourceHandler snmpResourceHandler = SNMPFactoryProvider.getFactory(version)
+                .createSNMPResourceHandler(snmpConfiguration)) {
+            final GetSNMPHandler getSNMPHandler = new GetSNMPHandler(snmpResourceHandler);
+            final SNMPSingleResponse response = getSNMPHandler.get(null, getFlowFileAttributesForSnmpGet(READ_ONLY_OID_1, READ_ONLY_OID_2));
+
+            assertEquals(READ_ONLY_OID_VALUE_1, response.getVariableBindings().get(1).getVariable());
+            assertEquals(READ_ONLY_OID_VALUE_2, response.getVariableBindings().get(0).getVariable());
+            assertEquals(SUCCESS, response.getErrorStatusText());
```

Comment:
We have a `Map` input and we still rely on an ordered result. I would do try to avoid that.
```suggestion
            Set<String> expectedVariables = new HashSet<>(Arrays.asList(READ_ONLY_OID_VALUE_1, READ_ONLY_OID_VALUE_2));
            Set<String> actualVariables = response.getVariableBindings().stream().map(SNMPValue::getVariable).collect(Collectors.toSet());

            assertEquals(expectedVariables, actualVariables);
            assertEquals(SUCCESS, response.getErrorStatusText());
```